src/Microsoft.FeatureManagement/ServiceCollectionExtensions.cs

```diff
@@ -27,31 +28,32 @@ public static IFeatureManagementBuilder AddFeatureManagement(this IServiceCollec
         {
             services.AddLogging();
 
+            services.AddMemoryCache();
+
             //
             // Add required services
             services.TryAddSingleton<IFeatureDefinitionProvider, ConfigurationFeatureDefinitionProvider>();
 
             services.AddSingleton(sp => new FeatureManager(
                 sp.GetRequiredService<IFeatureDefinitionProvider>(),
-                sp.GetRequiredService<IEnumerable<IFeatureFilterMetadata>>(),
-                sp.GetRequiredService<IEnumerable<ISessionManager>>(),
-                sp.GetRequiredService<ILoggerFactory>(),
-                sp.GetRequiredService<IOptions<FeatureManagementOptions>>(),
-                sp.GetRequiredService<IOptions<TargetingEvaluationOptions>>())
+                sp.GetRequiredService<IOptions<FeatureManagementOptions>>()?.Value)
             {
-                Configuration = sp.GetService<IConfiguration>(),
-                TargetingContextAccessor = sp.GetService<ITargetingContextAccessor>(),
+                FeatureFilters = sp.GetService<IEnumerable<IFeatureFilterMetadata>>(),
+                SessionManagers = sp.GetService<IEnumerable<ISessionManager>>(),
```

Comment:
`SessionManagers = sp.GetService<IEnumerable<ISessionManager>>(),`

FeatureManager throws if SessionManagers is set to null, but I thought `sp.GetService<IEnumerable<ISessionManager>>()` would return null if no-one registered any session managers?

Reply:
No, it wouldn't return null if there is not any registered ISessionManager. An empty IEnumerable will be returned.
Anyway, I agree with that we should change it into GetRequiredService<>